### DISCUSSION TITLE
Using Semantic Kernel v1.0 and greater with prompt flow in Azure AI Studio

### DISCUSSION THREAD

**Discussion by leeprtc** [microsoft/promptflow#3536] [Q&A] [answered]
Semantic Kernel v1.0 and greater Python packages require at least Python 3.10 or greater. When I add "semantic-kernel >= 1.0" to requirements.txt, I'm getting the following error.

ERROR: Ignored the following versions that require a different python version: 0.9.7b1 Requires-Python <3.13,>=3.10; 0.9.8b1 Requires-Python <3.13,>=3.10; 0.9.9b1 Requires-Python <3.13,>=3.10; 1.0.0 Requires-Python <3.13,>=3.10; 1.0.0rc1 Requires-Python <3.13,>=3.10; 1.0.1 Requires-Python <3.13,>=3.10; 1.0.2 Requires-Python <3.13,>=3.10; 1.0.3 Requires-Python <3.13,>=3.10; 1.0.4 Requires-Python <3.13,>=3.10; 1.0.5 Requires-Python <3.13,>=3.10; 1.1.0 Requires-Python <3.13,>=3.10; 1.1.1 Requires-Python <3.13,>=3.10; 1.1.2 Requires-Python <3.13,>=3.10; 1.2.0 Requires-Python <3.13,>=3.10
ERROR: Could not find a version that satisfies the requirement semantic-kernel>=1.0 (from versions: 0.0.1.dev0, 0.1.0.dev0, 0.2.0.dev0, 0.2.1.dev0, 0.2.2.dev0, 0.2.3.dev0, 0.2.4.dev0, 0.2.5.dev0, 0.2.6.dev0, 0.2.7.dev0, 0.2.8.dev0, 0.2.9.dev0, 0.3.0.dev0, 0.3.1.dev0, 0.3.2.dev0, 0.3.3.dev0, 0.3.4.dev0, 0.3.5.dev0, 0.3.6.dev0, 0.3.7.dev0, 0.3.8.dev0, 0.3.9.dev0, 0.3.10.dev0, 0.3.11.dev0, 0.3.12.dev0, 0.3.13.dev0, 0.3.14.dev0, 0.3.15.dev0, 0.4.0.dev0, 0.4.1.dev0, 0.4.2.dev0, 0.4.3.dev0, 0.4.4.dev0, 0.4.5.dev0, 0.4.6.dev0, 0.4.7.dev0, 0.5.0.dev0, 0.5.1.dev0, 0.9.0b1, 0.9.1b1, 0.9.2b1, 0.9.3b1, 0.9.4b1, 0.9.5b1, 0.9.6b1)
ERROR: No matching distribution found for semantic-kernel>=1.0

It does not seem like the default runtime is using Python 3.10 or greater. When can we expect the default runtime to be updated to run with Python 3.10 or greater? 

If I want to use Semantic Kernel v1.0 or greater, does this mean I need to create my own custom runtime? Prompt flow Visual Studio Code extension implies that prompt flow works with Python 3.9. Is it ok to run prompt flow with Python version greater than 3.9?

**Comment by D-W-** [comment, marked as answer]
Hi @leeprtc , for Azure AI Studio questions, please follow [this guide](https://microsoft.github.io/promptflow/cloud/azureai/faq.html#how-to-create-feedback-for-azure-portal-experience) to provide feedback. We currently only track open-source version of prompt flow here.

For your questions, we have a private feature to allow you build a private executor image with your dependency:

1  Follow this to build an image [promptflow-python-bundle · PyPI](https://pypi.org/project/promptflow-python-bundle/)
2. Put the built image in flow.dag.yaml: 
 
![image](https://github.com/microsoft/promptflow/assets/7776147/9e139637-dd06-4900-9bf3-7a18ec1df9cf)

3. Add &flight=PFMultiContainer in UX url.

**Comment by ebotiab** [comment]
This gives the following error:
{"level":"[WARN]","ts":"Jul  25 09:19:34","logger":"PromptFlow","caller":"util/http_util.go:90","msg":"Error calling endpoint: Get \"http://127.0.0.1:46451/aml-api/v1.0/version\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)"}
{"level":"[INFO]","ts":"Jul  25 09:19:34","logger":"PromptFlow.FlowHandler","caller":"handler/server.go:316","msg":"Request timeout, retrying...","request_id":"be468ef5-d6df-4159-8920-fa8a57ddf2cd","retry":4,"error":"Get \"http://127.0.0.1:46451/aml-api/v1.0/version\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)"}

**Reply by jakeatmsft** [reply]
any updates?  Working on semantic kernel + ai foundry agents and this is a blocking issue.
